From 4dbe4845c75318a491839c56cb6b43ef3c0b7cfa Mon Sep 17 00:00:00 2001
From: Junhao He <hejunhao3@huawei.com>
Date: Thu, 8 Dec 2022 21:45:23 +0800
Subject: [PATCH] coresight: ete: Add acpi match id for Hip09
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 4dbe4845c75318a491839c56cb6b43ef3c0b7cfa
Modified-by-SEL: No


driver inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I5YCYK

--------------------------------------------------------------------------

Add the acpi match id for Hip09 ETE

Signed-off-by: Junhao He <hejunhao3@huawei.com>
Reviewed-by: Ling Mingqiang <lingmingqiang@huawei.com>
Reviewed-by: Yicong Yang <yangyicong@huawei.com>
Reviewed-by: Xiongfeng Wang <wangxiongfeng2@huawei.com>
Reviewed-by: Yang Shen <shenyang39@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 drivers/hwtracing/coresight/coresight-etm4x-core.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/hwtracing/coresight/coresight-etm4x-core.c b/drivers/hwtracing/coresight/coresight-etm4x-core.c
index eec2ded50646..95c92d2d3915 100644
--- a/drivers/hwtracing/coresight/coresight-etm4x-core.c
+++ b/drivers/hwtracing/coresight/coresight-etm4x-core.c
@@ -2140,12 +2140,19 @@ static const struct of_device_id etm4_sysreg_match[] = {
 	{}
 };
 
+static const struct acpi_device_id static_ete_ids[] = {
+	{"HISI0461", 0},
+	{}
+};
+MODULE_DEVICE_TABLE(acpi, static_ete_ids);
+
 static struct platform_driver etm4_platform_driver = {
 	.probe		= etm4_probe_platform_dev,
 	.remove		= etm4_remove_platform_dev,
 	.driver			= {
 		.name			= "coresight-etm4x",
 		.of_match_table		= etm4_sysreg_match,
+		.acpi_match_table       = static_ete_ids,
 		.suppress_bind_attrs	= true,
 	},
 };
-- 
2.33.0

